packages/functions/src/dataTriggers.ts

```diff
@@ -461,3 +466,116 @@ export const createAttendedSlotOnAttendance = functions
 
     await batch.commit();
   });
+
+export const createCustomerStats = functions
+  .region(__functionsZone__)
+  .firestore.document(
+    `${Collection.Organizations}/{organization}/${OrgSubCollection.Bookings}/{secretKey}/${BookingSubCollection.BookedSlots}/{bookingId}`
+  )
+  .onWrite(async (_, context) => {
+    const { organization, secretKey } = context.params as Record<
+      string,
+      string
+    >;
+
+    const db = admin.firestore();
+
+    const bookingRef = db
+      .collection(Collection.Organizations)
+      .doc(organization)
+      .collection(OrgSubCollection.Bookings)
+      .doc(secretKey);
+
+    // Fetch the booking document
+    const bookingDoc = await bookingRef.get();
+    const { id: customerId } = bookingDoc.data() as CustomerBookings;
+
+    // Fetch documents from a subcollection of the booking
+    const bookedSlotsRef = bookingRef.collection("bookedSlots");
+    const bookedSlotsSnapshot = await bookedSlotsRef.get();
+
+    const bookedSlots: DocumentData = {};
+    bookedSlotsSnapshot.forEach((doc) => {
+      Object.assign(bookedSlots, { [doc.id]: doc.data() });
```

Comment:
Is there a reason not to write this as `bookedSlots[doc.id] = doc.data();`?

Reply:
No, yours is more readable, actually. They do the same thing but mine was just out of habit.